lib/engine/game/g_1880/game.rb

```diff
@@ -321,6 +322,41 @@ def event_signal_end_game!
           game_end_check
         end
 
+        def event_can_buy_trains!
+          @log << "-- Event: #{EVENTS_TEXT['can_buy_trains'][1]} --"
+          @can_buy_trains = true
+        end
+
+        def force_exchange_rocket
+          return unless @rocket
+          return if @rocket.closed?
+
+          player_corps = @corporations.select { |c| c.owner == @p7.owner }
+          corps_with_slots = player_corps.select { |c| c.trains.length < train_limit(c) }
+          if player_corps.empty?
+            @rocket.close!
+          elsif corps_with_slots.empty?
```

Comment:
They should be able to acquire and then discard another train

> Forced exchange: ...If at this point the owner of P7 has no company with a free space for a train because of the train
limit, he can voluntarily surrender an existing train in order to create a space for the 4-train (this is
the only possible time in the game that a train can be voluntarily surrendered).

Reply:
a rule detail that was lost when i wrote up the task list. Good catch

Reply:
fixed the entire assign rocket step including this code. 
It will only export train if rocket owner owns no corps, otherwise an immediate assign followed by train drop if needed